netdata/netdata
# Add support for alert notifications to ntfy.sh

##### Summary
This PR adds the option to send push notifications to phones and/or PCs via PUT/POST requests, using https://ntfy.sh.

Closes https://github.com/netdata/netdata/issues/13432  (thanks to @c0depool and @skibbipl for kickstarting the work!). 

##### Test Plan

Using the branch of this PR:

1. Ensure `SEND_NTFY="YES"` in `health_alarm_notify.conf`.
2. Configure `DEFAULT_RECIPIENT_NTFY=` in `health_alarm_notify.conf` with one or more recipients, such as: 
```
DEFAULT_RECIPIENT_NTFY="https://ntfy.sh/netdata-example-topic-1456 https://ntfy.sh/netdata-example-topic-1457"
```

3. Subscribe to the above topics using [the phone app](https://docs.ntfy.sh/subscribe/phone/), [the web app](https://docs.ntfy.sh/subscribe/web/) or [the cli](https://docs.ntfy.sh/subscribe/cli/).
4. Trigger an alert and check that it is received (and cleared) properly.

##### Additional Information


<details> <summary>For users: How does this change affect me?</summary>
It adds support for sending `ntfy.sh` notifications, in a pub/sub way.
</details>
